std/mem.zig

```diff
@@ -84,6 +84,36 @@ pub const Allocator = struct {
     }
 };
 
+pub var c_allocator = Allocator {
+    .allocFn = cAlloc,
+    .reallocFn = cRealloc,
+    .freeFn = cFree,
+};
+
+const c = @cImport(@cInclude("stdlib.h"));
+
+fn cAlloc(self: &Allocator, n: usize, alignment: usize) -> %[]u8 {
+    if (c.malloc(usize(n))) |mem| {
+        @ptrCast(&u8, mem)[0..n]
+    } else {
+        error.OutOfMemory
+    }
+}
+
+fn cRealloc(self: &Allocator, old_mem: []u8, new_size: usize, alignment: usize) -> %[]u8 {
+    const old_ptr = @ptrCast(&c_void, &old_mem[0]);
+    if (c.realloc(old_ptr, usize(new_size))) |mem| {
+        @ptrCast(&u8, mem)[0..new_size]
+    } else {
+        error.OutOfMemory
+    }
+}
+
+fn cFree(self: &Allocator, old_mem: []u8) {
+    const old_ptr = @ptrCast(&c_void, &old_mem[0]);
```

Comment:
I think it makes more sense to use `old_mem.ptr` instead of `&old_mem[0]`

Reply:
Always forget this exists. Is there a list of member values for each type listed anywhere right now? For example `len` on slices, `bit_count` on types etc. Don't think I've noticed anything specifically in the documentation.

Reply:
hmm I think that is one of the document TODOs